network/common/src/main/java/org/apache/spark/network/TransportContext.java

```diff
@@ -99,9 +108,9 @@ public TransportServer createServer() {
    * be used to communicate on this channel. The TransportClient is directly associated with a
    * ChannelHandler to ensure all users of the same channel get the same TransportClient object.
    */
-  public TransportChannelHandler initializePipeline(SocketChannel channel) {
+  public TransportChannelHandler initializePipeline(SocketChannel channel, RpcHandler rpcHandler) {
```

Comment:
The rpcHandler vs appRpcHandler thing is definitely confusing to someone reading this for the first time, please add a comment here or in TransportContext about the difference.